pkg/libvirt/resolve_vm.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package libvirt
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+)
+
+const (
+	libvirtPath string = "/var/run/libvirt/qemu/"
+)
+
+func getThreadIDsForPID(pid string) []string {
+	threadIDs := []string{}
+
+	procDir := fmt.Sprintf("/proc/%s/task", pid)
+	files, err := ioutil.ReadDir(procDir)
+	if err != nil {
+		return nil
+	}
+
+	for _, file := range files {
+		threadIDs = append(threadIDs, file.Name())
+	}
+
+	return threadIDs
+}
+
+func GetCurrentVMPID() (map[string]string, error) {
```

Comment:
can you write a unit test for this function?

Reply:
done